sklearn/ensemble/gradient_boosting.py

```diff
@@ -1846,3 +1853,466 @@ def apply(self, X):
+
+            * ``parameters``, a dict of parameter settings
+            * ``mean_validation_score``, the mean score over the
+              cross-validation folds
+            * ``cv_validation_scores``, the list of scores for each fold
+
+    best_score_tuple_ : named tuple
+        The attributes of the model which gave the best score.
+
+        * ``parameters``, a dict of parameter settings, which includes the
+          `n_estimators` parameter
+        * ``mean_validation_score``, the mean score over the
+          cross-validation folds
+        * ``cv_validation_scores``, the list of scores for each fold
+
+    best_params_ : dict
+        The parameters which gave the best score, along with the `n_estimators`
+        parameter.
+
+    best_estimator_ : GradientBoostingClassifier
+        If `refit` was `True`, the model with the best score fit to the entire
+        dataset.
+
+    """
+
+    _estimator_class = GradientBoostingClassifier
+
+    def __init__(self, n_stop_rounds=10, score_precision=2,
+                 max_iterations=10000, cv=3, scoring=None, refit=True,
+                 n_jobs=1, pre_dispatch='2*n_jobs', verbose=0, loss='deviance',
+                 learning_rate=0.1, subsample=1.0, min_samples_split=2,
+                 min_samples_leaf=1, min_weight_fraction_leaf=0., max_depth=3,
+                 init=None, random_state=None, max_features=None,
+                 max_leaf_nodes=None, presort='auto'):
+
+        self.n_stop_rounds = n_stop_rounds
+        self.score_precision = score_precision
+        self.max_iterations = max_iterations
+        self.cv = cv
+        self.scoring = scoring
+        self.refit = refit
+        self.n_jobs = n_jobs
+        self.pre_dispatch = pre_dispatch
+        self.verbose = verbose
+
+        self.loss = loss
+        self.learning_rate = learning_rate
+        self.subsample = subsample
+        self.min_samples_split = min_samples_split
+        self.min_samples_leaf = min_samples_leaf
+        self.min_weight_fraction_leaf = min_weight_fraction_leaf
+        self.max_depth = max_depth
+        self.init = init
+        self.random_state = random_state
+        self.max_features = max_features
+        self.max_leaf_nodes = max_leaf_nodes
+        self.presort = presort
+
+    def fit(self, X, y):
+        """Run fit with all sets of parameters till convergence.
+    
+        X : array-like, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples
+            and n_features is the number of features.
+        y : array-like, shape = [n_samples]
+            Target values (integers in classification, real numbers in
+            regression)
+            For classification, labels must correspond to classes.
+        """
+    
+        params = self._get_params()
+        X, y = check_X_y(X, y, dtype=DTYPE)
+    
+        cv = check_cv(self.cv, y)
+        param_iter = ParameterGrid(params)
```

Comment:
What about also accepting an additional constructor parameter named "n_param_combo=None" by default to use exhaustive search with `ParameterGrid` by default but randomized parameter search with `ParameterSampler` when the user passes some integer value?